htdocs/compta/facture/class/facture.class.php

```diff
@@ -4433,18 +4433,18 @@ public function setDiscount($user, $remise, $notrigger = 0)
 	{
 		// Clean parameters
 		if (empty($remise)) {
-			$remise = 0;
+			$remise = 0.0;
 		}
 
 		if ($user->hasRight('facture', 'creer')) {
-			$remise = price2num($remise, 2);
+			$remise = (float) price2num($remise, 2);
 
 			$error = 0;
 
 			$this->db->begin();
 
 			$sql = 'UPDATE '.MAIN_DB_PREFIX.'facture';
-			$sql .= ' SET remise_percent = '.((float) $remise);
+			$sql .= ' SET remise_percent = '.$remise;
```

Comment:
When we forge a sql request, we must always for the escaping or the cast for numeric value.